src/logic/activities.js

```diff
@@ -299,6 +299,38 @@ export const roundsMissingScorecards = wcif =>
     .filter(round => groupActivitiesAssigned(wcif, round.id))
     .filter(round => parseActivityCode(round.id).eventId !== '333fm');
 
+export const competitorsRegisteredForAnEventWithoutGroups = wcif => {
+  let persons = [];
+  wcif.persons.forEach(person => {
+    if (!person.registration || person.registration?.status !== 'accepted')
+      return;
+    person.registration.eventIds.forEach(eventId => {
+      const hasCompetitorAssignment = person.assignments.some(
+        a =>
+          a.assignmentCode === 'competitor' &&
+          activityById(wcif, a.activityId).activityCode.startsWith(
+            `${eventId}-r1`
+          )
+      );
+      if (!hasCompetitorAssignment) {
+        if (!persons.some(p => p.person.wcaUserId === person.wcaUserId)) {
+          persons.push({ person, eventIds: [eventId] });
+        } else {
+          let newPerson = persons.find(
+            p => p.person.wcaUserId === person.wcaUserId
+          );
+          newPerson.eventIds.push(eventId);
+          persons = persons.filter(
+            p => p.person.wcaUserId !== person.wcaUserId
+          );
+          persons.push(newPerson);
```

Comment:
`.eventIds.push(eventId)` above modifies the existing person object in the list, so we don't need to remove+push it again:

```suggestion
```